apis/logging/v1/output_types.go

```diff
@@ -304,3 +307,28 @@ type Http struct {
 	// +optional
 	Schema string `json:"schema,omitempty"`
 }
+
+type AzureMonitor struct {
+	//CustomerId che unique identifier for the Log Analytics workspace.
+	//https://learn.microsoft.com/en-us/azure/azure-monitor/logs/data-collector-api?tabs=powershell#request-uri-parameters
+	CustomerId string `json:"customerId,omitempty"`
+
+	//LogType the record type of the data that is being submitted.
+	//Can only contain letters, numbers, and underscores (_), and may not exceed 100 characters.
+	//https://learn.microsoft.com/en-us/azure/azure-monitor/logs/data-collector-api?tabs=powershell#request-headers
+	LogType string `json:"logType,omitempty"`
```

Comment:
Is there/could there be and relationship between our  logType [applicaiton, infrastructure, audit] and Azure's logType? E.g. would it be reasonable to default the Azure logType from ours if the user doesn't specify one? If not, need to explain that they are unreleated.

Reply:
@alanconway Azure LogType not related to our notation. 
In Azure Monitor Logs, the Log Type is a property that helps categorize and organize log data based on the type of data being collected or ingested. It essentially represents the schema or structure of the data being logged. Log Type is used to identify the kind of logs being sent to Azure Monitor and is often specified as a part of log data ingestion.

When sending logs to Azure Monitor, you must specify the Log Type to define the format and structure of the log data. This helps Azure Monitor understand how to parse and process the incoming data and enables users to effectively query and analyze the logs based on their specific types.

Reply:
This sounds the same as what we would consider as LogType.  User's define it to something that makes sense to them.  I would argue this is an optional setting that we default to the values we already use: app, infra, audit.  user's override it as necessary and likely can additionally templatize it (e.g. set it to a namespace value based on whats in the record)

Reply:
Vector don't support template syntax `log_type`

Reply:
Agree that we can make this field optional